### ISSUE TITLE
Discord auth integration

### ISSUE THREAD

**Issue by warriorsoul15** [eveseat/seat#104]
Discord is starting to be used pretty well by EVE Players in small to mid-sized corps.  Its api was recently released.  For the unfamiliar its a more gaming friendly slack with voice features.  Its API is released and I have seen homebrew 'auth to discord' integrations (not for seat) that manage invites to the Discord as well as setting permissions on channels according to 'auth groups'.   as well as kicking users from discord if they leave corp/alliance.

**Comment by jagyro** [comment]
+1

**Comment by CokkocZateki** [comment]
+1

**Comment by Cynabal** [comment]
+1

**Comment by warriorsoul15** [comment]
+100

**Comment by op6sie** [comment]
+1

**Comment by RichterSR** [comment]
Here (https://github.com/R4stl1n/allianceauth) is an other great EVE Corp/Ally Administration tool. Maybe this can help here.

**Comment by warlof** [comment]
Is anybody know if Discord provide a way to know when channel are created ?
Slack provide [RTM](https://api.slack.com/rtm) and more recently [Event API](https://api.slack.com/events-api) which provide that kind of thing. The main purpose is to keep channels up to date without firing a stalker job every minutes. Plus, it ensure that data are instantly up to date.

**Comment by warriorsoul15** [comment]
https://discordapp.com/developers/docs/topics/gateway#events

**Comment by warlof** [comment]
this "Sent when a new channel is created, relevant to the current user" is fearing me but will try :)
Thank for link

**Comment by Bingmano** [comment]
+1

**Comment by G0wd** [comment]
+1

**Comment by dae-eliador** [comment]
+1

**Comment by shibdib** [comment]
https://github.com/shibdib/Dramiel shameless plug, bot will auth people to groups via sso

**Comment by leonjza** [comment]
This will be natively supported in SeAT soon.

**Comment by monasucks** [comment]
+1

**Comment by shadowcodex** [comment]
@leonjza any update on the native support?

**Comment by apfelqoo** [comment]
+1 :)

**Comment by leonjza** [comment]
Alrighty so quick update on this one. Admittedly, things have slowed down a little in SeAT world. Primarily, this is due to uncertainties about the changes that are needed to make SeAT and ESI happy. SeAT will be making full use of ESI soon, and work has already started by building a standalone ESI client library for it (which can be seen [here](https://github.com/eveseat/eseye)).

Why is this relevant to Discord/(any auth for that matter) integration? The main reason is due to the way account registrations/character linkages will change. Currently, a SeAT account and EVE API keys are loosely coupled. With ESI and the move to use SSO tokens to gain access to API data, the current implementation will need a little bit of rework as there is going to be a much tighter coupling between a SeAT/SSO account and its characters.

TL;DR. Hang tight. ;)

**Comment by warlof** [comment]
In addition, I'll perhaps take back my sync bot between discord and seat since I'm starting to understand a bit more their API.

Things are a bit different though, since Slack (at least the free plan) is user based whereas discord is channel based (we don't know when an user join or leave a channel, but we know when a channel is modified. Therefore, we're able to check if permissions has changed).

But, this will not start before SeAT 3.x I think which will use full ESI and SSO (@leonjza ?).
And, as him said, due to those change, authentication and user meaning will perhaps change (a lot).

**Comment by ErickAsmock** [comment]
+1

**Comment by H4Z4RD-H42** [comment]
For now i'm using [Keepstar](https://github.com/shibdib/Keepstar) (the continue of Dramiel) and it work very well, but if SeAT add this features, i drop Keepstar in favor of it :D

**Comment by warlof** [comment]
to be close -> https://github.com/warlof/seat-discord-connector

**Comment by ppfeufer** [comment]
As far as I can gather, SeAT is an Audit not an Auth.

If you want to have authentication for external services like Discord, Mumble or forums, I suggest using an Auth system that is specialized in doing exactly that.

https://gitlab.com/allianceauth/allianceauth for example does a very good job here. Pure authentication system for external services with a couple of neat features like timerboard, SRP management, group management for the external services, application management and so on.

The developer of AllianceAuth is currently looking into integrating SeAT as a service again into AllianceAuth as it was before the ESI changes. Would be nice to see both developers working together to make that happen instead of duplication functions that already exists.

Just my 2 ISk here.

**Comment by herpaderpaldent** [comment]
^ don't listen to this guy. There are plugins for Discourse, Slack, Discord, SRP-Module and Fitting/Doctrins out there. Take a look into the docs: https://eveseat.github.io/docs/community_packages/#package-list

**Comment by warlof** [comment]
Since Discord connector has been released, I think this issue can be close ?

**Comment by herpaderpaldent** [comment]
Yeah this should be closed

**Comment by warlof** [comment]
Closing due to inactivity and wishlist covered by third party package.
